lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,191 @@
+var edx = edx || {},
+
+    Dropdown = (function() {
+
+        var dropdown = {
+
+            opts: {
+                page: $(document),
+                midpoint: ($(window).width() / 2),
+                button: $('.button-more.has-dropdown'),
+                button_active: 'is-active',
+                menu: $('.dropdown-menu'),
+                menu_inactive: 'is-hidden',
+                menu_active: 'is-visible',
+                menu_align: 'align-',
+                text_align: 'control-text-align'
+            },
+
+            init: function(parent) {
+
+                if (dropdown.opts.button.length && dropdown.opts.menu.length) {
+
+                    if (parent) {
+                        dropdown.opts.page = $(parent);
+                    }
+
+                    dropdown.listenForClick();
+                    dropdown.listenForKeypress();
+                }
+            },
+
+            listenForClick: function() {
+
+                dropdown.opts.button.on('click', function() {
+                    dropdown.closeDropdownMenus(); // close any open menus
+                    dropdown.openDropdownMenu($(this)); // then open the chosen menu
+                });
+
+                dropdown.opts.page.on('click', function() {
+                    dropdown.closeDropdownMenus();
+                });
+            },
+
+            handlerIsAction: function(key, menu) {
```

Comment:
BTW, the code is now much easier to read. Thanks for making these helper methods!